easybuild/easyconfigs/j/jax/jax-0.4.24-foss-2023a-CUDA-12.1.1.eb

```diff
@@ -0,0 +1,120 @@
+# note: this *must* be the exact same commit as used in third_party/{xla,"other"}/workspace.bzl
+local_xla_commit = '12eee889e1f2ad41e27d7b0e970cb92d282d3ec5'
+local_tfrt_commit = '4665f7483063a16b6113a05eb45f98103cc1d611'
+local_repo_opt = '--bazel_options="--override_repository=xla=%%(builddir)s/xla-%s" ' % local_xla_commit
+local_repo_opt += '--bazel_options="--override_repository=runtime=%%(builddir)s/tf_runtime-%s" ' % local_xla_commit
+
+local_test = "NVIDIA_TF32_OVERRIDE=0 CUDA_VISIBLE_DEVICES=0 "
+local_test += "XLA_PYTHON_CLIENT_ALLOCATOR=platform "
+local_test += "JAX_ENABLE_X64=true pytest -vv tests "
+
+use_pip = True
+
+default_easyblock = 'PythonPackage'
+default_component_specs = {
+    'sources': [SOURCE_TAR_GZ],
+    'source_urls': [PYPI_SOURCE],
+    'start_dir': '%(name)s-%(version)s',
+    'use_pip': True,
+    'sanity_pip_check': True,
+    'download_dep_fail': True,
+}
+
+components = [
+    ('absl-py', '2.1.0', {
+        'options': {'modulename': 'absl'},
+        'checksums': ['7820790efbb316739cde8b4e19357243fc3608a152024288513dd968d7d959ff'],
+    }),
+    ('jaxlib', version, {
+        'sources': [
+            '%(name)s-v%(version)s.tar.gz',
+            {
+                'download_filename': '%s.tar.gz' % local_xla_commit,
+                'filename': 'xla-%s.tar.gz' % local_xla_commit,
+            },
+            {
+                'download_filename': '%s.tar.gz' % local_tfrt_commit,
+                'filename': 'tf_runtime-%s.tar.gz' % local_tfrt_commit,
+            },
+        ],
+        'source_urls': [
+            'https://github.com/google/jax/archive/',
+            'https://github.com/tensorflow/runtime/archive',
+            'https://github.com/openxla/xla/archive'
+        ],
+        'patches': [
+            ('jax-0.4.24_xla-%s_indexing_analysis_small_vector.patch' % local_xla_commit[:7],
+             '../xla-%s' % local_xla_commit),
+            # cuda-noncanonical-include-paths still required?: 
+            # ('jax-0.4.24_xla-%s_cuda-noncanonical-include-paths.patch' % local_xla_commit[:7],
+            # '../xla-%s' % local_xla_commit),
+        ],
+        'checksums': [
+            {'jaxlib-v0.4.24.tar.gz':
+             'c4e6963c2c36f634a9a1765e476a1ed4e6c4a7954465ebf72e29f344c28ddc28'},
+            {'xla-12eee889e1f2ad41e27d7b0e970cb92d282d3ec5.tar.gz':
+             'db007b6628cfe108c63f45d611c6de910abe3ee827e55f08314ce143c4887d66'},
+            {'tf_runtime-4665f7483063a16b6113a05eb45f98103cc1d611.tar.gz':
+             '3aa0ab30fe94dab33f20824b9c2d8e7c3b6017106c833b12070f71d2e0f1d6d6'},
+            {'jax-0.4.24_xla-12eee88_indexing_analysis_small_vector.patch':
+             '7187cdd08cce12d0af889494317cb8c32865487d1d6d9254064cb62fd3453b6d'},
+        ],
+        'start_dir': 'jax-jaxlib-v%(version)s',
+        'buildopts': local_repo_opt
+    }),
+]
+
+exts_list = [
+    (name, version, {
+        # 'patches': ['jax-0.4.24_cusparse.patch'],
```

Comment:
are any of the old patches still relevant?
```
jax-0.3.23_relax-testPoly5-tolerance.patch
jax-0.3.25_skip-qdwh-test-rank-deficient-deficient.patch
jax-0.3.25_skip-test-bcoo-dot-general-cusparse.patch
jax-0.4.4_cusparse.patch
```

Reply:
* `jax-0.4.4_cusparse.patch` - this should be safe to remove. Bug seen in CUDA 11.7.0 but not in 11.7.1. See https://github.com/google/jax/issues/14663
* `jax-0.3.25_skip-test-bcoo-dot-general-cusparse.patch` - the comment in this patch also points at the issue from the previous one
* `jax-0.3.25_skip-qdwh-test-rank-deficient-deficient.patch` - the [bug this patch points](https://github.com/google/jax/issues/14793) to is marked as fixed
* `jax-0.3.23_relax-testPoly5-tolerance.patch` - I added this in #16455 - I've not had this test fail in this PR.